test/integration/crcsuite/crcsuite.go

```diff
@@ -25,33 +25,176 @@ package crcsuite
 
 import (
 	"fmt"
+	"os"
+	"os/user"
+	"path/filepath"
 
 	"github.com/DATA-DOG/godog"
-	"github.com/code-ready/clicumber/testsuite"
+	"github.com/DATA-DOG/godog/gherkin"
+
+	clicumber "github.com/code-ready/clicumber/testsuite"
 )
 
-var sh testsuite.ShellInstance
+// FeatureContext defines godog.Suite steps for the test suite.
+func FeatureContext(s *godog.Suite) {
 
-func crcStateIs(expected string) error {
+	// CRC related steps
+	s.Step(`^removing CRC home directory succeeds$`,
+		RemoveCRCHome)
+
+	// CRC file operations
+	s.Step(`^file "([^"]*)" exists in CRC home folder$`,
+		FileExistsInCRCHome)
+	s.Step(`"(JSON|YAML)" config file "(.*)" in CRC home folder (contains|does not contain) key "(.*)" with value matching "(.*)"$`,
+		ConfigFileInCRCHomeContainsKeyMatchingValue)
+	s.Step(`"(JSON|YAML)" config file "(.*)" in CRC home folder (contains|does not contain) key "(.*)"$`,
+		ConfigFileInCRCHomeContainsKey)
+	s.Step(`removing file "(.*)" from CRC home folder succeeds$`,
+		DeleteFileFromCRCHome)
+
+	s.BeforeSuite(func() {
+
+		// delete existing crc instance (if any)
+		err := DeleteCRC()
+		if err != nil {
+			fmt.Println(err)
+		}
+
+		// remove $HOME/.crc
+		err = RemoveCRCHome()
+		if err != nil {
+			fmt.Println(err)
+		}
+	})
+
+	s.AfterFeature(func(this *gherkin.Feature) {
+		err := DeleteCRC()
+		if err != nil {
+			fmt.Println(err)
+		}
+	})
+
+}
 
-	fmt.Println("checking state of CRC...")
-	err := testsuite.ExecuteCommand("crc state")
+func DeleteFileFromCRCHome(fileName string) error {
 
+	usr, err := user.Current()
 	if err != nil {
 		return err
 	}
+	theFile := filepath.Join(usr.HomeDir, ".crc", fileName)
+
+	_, err = os.Stat(theFile)
+
+	if os.IsNotExist(err) {
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("file %s neither exists nor doesn't exist, error: %v", theFile, err)
```

Comment:
what does that even mean? `file <> neither exists nor doesn't exist` ?

Reply:
@praveenkumar It was a placeholder for an error message that I was supposed to think through later - we don't know what the error is, so the message has to be meaningless. But I agree it can be less annoying than that :). What should I change it to? And thanks for catching this!

Reply:
I don't think you need that `else` part at all, you need to catch if file is exist or not.

```golang
if _, err := os.Stat("/path/to/whatever"); os.IsNotExist(err) {
	// path/to/whatever does not exist
}
```

Reply:
I don't think you need that `else` part at all, you need to catch if file is exist or not.

```golang
if _, err := os.Stat("/path/to/whatever"); os.IsNotExist(err) {
	// path/to/whatever does not exist
}
```